drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,373 @@
+            yield orjson.loads(line.strip())
+        except orjson.JSONDecodeError:
+            # sometimes Drakvuf reports bad method names and/or malformed JSON
+            logger.debug("bad drakvuf log line: %s", line)
+
+
+def dynamic_capa_analysis(
+    analysis_dir: Path,
+    rules: RuleSet,
+    malware_pids: Optional[List[int]] = None,
+) -> Tuple[Path, crd.MatchResults]:
+
+    # save all api calls and native calls into one list which gets sorted by capa later on.
+    calls = []
+
+    # read api calls
+    if (analysis_dir / "apimon.log").exists():
+        with (analysis_dir / "apimon.log").open("r", errors="ignore") as apimon_fd:
+            calls += list(decode_json_lines(apimon_fd))
+    else:
+        logger.debug("missing apimon.log file")
+
+    # read syscalls
+    if (analysis_dir / "syscall.log").exists():
+        with (analysis_dir / "syscall.log").open("r", errors="ignore") as syscall_fd:
+            calls += list(decode_json_lines(syscall_fd))
+    else:
+        logger.debug("missing syscall.log file")
+
+    if not calls:
+        raise RuntimeError(
+            "both syscall.log and apimon.log are either empty or non-existent"
+        )
+
+    # initialize the Drakvuf capa feature extractor with the captured calls
+    extractor = get_drakvuf_feature_extractor(calls)
+
+    # apply a filter to the extractor if we wish to only analyze a specific set of processes
+    if malware_pids:
+        extractor = ProcessFilter(extractor, malware_pids)
+
+    # extract dymamic capabilities
+    capabilities, _ = find_capabilities(rules, extractor)
+
+    # analysis_dir is returned for conformity reasons with the ttp generation function (same routine for both static and dynamic capability extraction)
+    return analysis_dir, capabilities
+
+
+def get_process_memory_dumps(analysis_dir: Path, pid: int) -> Iterator[str]:
+    """get the memory dumps from a specific process."""
+    # this is used mainly to get the memdumps by a malware process
+    with (analysis_dir / "memdump.log").open("r") as f:
+        for line in f:
+            dump = orjson.loads(line)
+            if dump["PID"] == pid:
+                yield dump["DumpFilename"]
+
+
+def static_capa_analysis(
+    dump_path: Path, rules: RuleSet
+) -> Tuple[Path, crd.MatchResults]:
+
+    """get the input file's capa format"""
+    try:
+        input_format = get_auto_format(dump_path)
+    except UnsupportedFormatError:
+        logger.debug("dump %s has an unsupported format", dump_path)
+        return
+
+    # get the input file's adequate analysis backend
+    if input_format == FORMAT_DOTNET:
+        backend = BACKEND_DOTNET
+    else:
+        backend = BACKEND_VIV
+
+    # get the right capa feature extractor for the input: Vivisect, .NET, etc.
+    extractor = get_extractor(
+        dump_path,
+        input_format,
+        OS_WINDOWS,
+        backend,
+        [],
+        False,
+        False,
+        dump_path,
+    )
```

Comment:
If there are multiple arguments, I would use keyword arguments to improve readability. Especially if some of them have defaults and are defined as keyword arguments. https://github.com/mandiant/capa/blob/master/capa/loader.py#L189
```suggestion
    extractor = get_extractor(
        dump_path,
        input_format,
        OS_WINDOWS,
        backend=backend,
        sigpaths=[],
        should_save_workspace=False,
        disable_progress=False,
        sample_path=dump_path,
    )
```